src/App.tsx

```diff
@@ -15,48 +15,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetical = 'alphabet',
+  Length = 'length',
+  All = '',
+}
+
+interface SortProps {
+  sortGoods: SortType;
+  reversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortGoods, reversed }: SortProps,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortGoods) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortGoods) {
+        case SortType.Alphabetical:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortGoods, setSortGoods] = useState(SortType.All);
+  const [reversed, setReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortGoods,
+    reversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${sortGoods === SortType.Alphabetical ? '' : 'is-light'}`}
```

Comment:
To set class conditionally it's better to use classnames library https://www.npmjs.com/package/classnames